src/shape/shape.rs

```diff
@@ -150,8 +150,55 @@ pub enum TypedShape<'a> {
     #[cfg(feature = "dim2")]
     #[cfg(feature = "std")]
     RoundConvexPolygon(&'a RoundConvexPolygon),
-    /// A custom user-defined shape with a type identified by a number.
-    Custom(u32),
+    /// A custom user-defined shape.
+    Custom(&'a dyn Shape),
```

Comment:
Can a `dyn Shape` serialize/deserialize  ?

Reply:
@Vrixyz This is difficult due to the `Deserialize` trait bound having a lifetime. This is generally worked around using `erased_serde` but some additional steps are needed to make deserialization actually work. This sounds out of the scope of this PR.